### PR TITLE
Fix/jenkins

### LINKED ISSUE #102: setup Jenkins OAuth authentication access control

### LINKED ISSUE #103: extend Jenkins Docker image to restore from backup on start if necessary

### LINKED ISSUE #104: setup Jenkins to run as k8 deployment

### PR DESCRIPTION
**Pull request by frickjack**
* setup k8s deployment, public service, persistent volume, secrets, role, and service-account for jenkins, and updated .sh scripts accordingly
* extended jenkins Docker image with packer and kubectl
* setup jenkins startup script to auto-restore from nightly S3 backup if necessary
* add slack posts to Jenkinsfile pipelines
* add POC 'KubeCheck/Jenkinsfile' pipeline that issues kubectl commands
* patch terraform tf_files/aws/ to copy and encrypt the public ubuntu16-*-1.0.0-* AMI's that the packer scripts under uc-cdis/images repository now publish under the 'cdistest' AWS account 
* update README.md with some notes on RBAC and service accounts in k8s
* add Reuben and Rudy Google accounts to list of default logins for commons
* add 'revisionHistoryLimit: 2' to k8s deployments to hint the garbage collector to reclaim old replica sets
* update k8s versions in kube-up.sh

resolve #102 
resolve #103 
resolve #104

### CODE REVIEW

File: tf_files/aws/cloud.tf [hunk 1]
```diff
@@ -275,8 +275,55 @@ resource "aws_db_subnet_group" "private_group" {
...
 
+data "aws_ami" "public_login_ami" {
+  most_recent      = true
+
+  filter {
+    name   = "name"
+    values = ["ubuntu16-client-1.0.0-*"]
+  }
+
+  owners     = ["707767160287"]
```
Comment (philloooo) [inline tf_files/aws/cloud.tf]: same as images pr, let's change it to variable

File: kube/services/jenkins/jenkins-deploy.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            valueFrom:
+              configMapKeyRef:
+                name: global
+                key: hostname
+          - name: AWS_DEFAULT_REGION
+            value: us-east-1
+          - name: AWS_ACCESS_KEY_ID
+            valueFrom:
+              secretKeyRef:
+                name: jenkins-secret
```
Comment (philloooo) [inline kube/services/jenkins/jenkins-deploy.yaml]: where is this?

Comment (frickjack) [reply kube/services/jenkins/jenkins-deploy.yaml]: ugh - looks like I just created that secret by hand - I'll add some automation :-p
kubectl create secret generic jenkins-secret "--from-literal=aws_access_key_id=$aws_access_key_id" "--from-literal=aws_secret_access_key=$aws_secret_access_key"

File: tf_files/configs/kube-services.sh [hunk 3]
```diff
@@ -52,10 +56,36 @@ kubectl apply -f services/indexd/indexd-service.yaml
 kubectl apply -f services/gdcapi/gdcapi-service.yaml
 ./services/revproxy/apply_service
 
+if [ "$KUBE_JENKINS" = "enabled" ]; then
```
Comment (zflamig) [inline tf_files/configs/kube-services.sh]: These files get rendered by Terraform so the variables are converted out... need to escape them or turn them into TF variables?

Comment (philloooo) [reply tf_files/configs/kube-services.sh]: nice catch :)

**Pull request merged**